lib/phobos/listener.rb

```diff
@@ -123,7 +123,14 @@ def stop_listener
     end
 
     def start_consumer_loop
-      @delivery == 'batch' ? consume_each_batch : consume_each_message
+      # validate batch handling
+      if @delivery == 'batch'
+        consume_each_batch
+      elsif handler_class.method_defined?(:consume)
```

Comment:
This is a bit complicated for me. I would be open to have a new delivery option `batch_inline` since I think this makes the readme section more clear. Example `phobos.yml`:

```
listeners:
  - handler: MyOldBatchHandler
     delivery: batch
  - handler: MyNewBatchHandler
     delivery: batch_inline
```

When opening `MyNewBatchHandler` I would expect to see `include Phobos::BatchHandler`, and when opening `MyOldBatchHandler` I would expect to see the old classic `include Phobos::Handler`.

This would make this `if` block much clearer and more [#batch_inline]

Of course, the delivery option `batch_inline` must be documented in the README but I hope you agree this will make adoption much easier for other people.